src/dodal/devices/i22/fswitch.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+from enum import Enum
+from typing import Dict
+
+from bluesky.protocols import Reading
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.device import DeviceVector
+from ophyd_async.epics.signal import epics_signal_r
+
+
+class FilterState(str, Enum):
+    IN = "IN"
+    OUT = "OUT"
+
+
+class FSwitch(StandardReadable):
+    """
+    Device for i22's fswitch. This currently only implements the minimum
+    functionality for retrieving the number of lenses inserted.
+
+    Eventually this should be combined with the transfocator device in the i04
+    module but is currently incompatible as the Epics interfaces are different.
```

Comment:
Should: I would prefer this as a Github issue rather than a docstring, could just link to the issue here

Reply:
Agreed, as discussed at https://github.com/DiamondLightSource/dodal/issues/399#issuecomment-2049824696 I think the first thing to do to combine them would be to reach out to i22 and see if they want the logic for beamsize calculation. @aragaod (i04 beamline scientist) has said he'd be happy to chat to them about how i04 use it if we arrange the discussion.

Reply:
> new version contains a PV that provides this value already

I presume controls will check the maths with i22 scientists directly? Very much wouldn't assume that i04's maths necessarily just works for i22...

Reply:
I'm assuming that for the number of lenses in the beam it is just counting up the filter PVs that are "IN_BEAM" as this device is doing? Or is there some complexity I'm missing?